### PR TITLE
refactor: refactor dom tooltip of vchart

### PR DESCRIPTION
**Pull request by xile611**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

1. dom-tooltip 渲染的时候，不再走：spec转换到canvas-tooltip 属性，再转换到dom属性这个流程，直接根据spec 生成dom内容
2. 去除对dom节点的封装，直接使用dom inner html
3. dom-tooltip所有类名增加统一的前缀，避免冲突

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/plugin/components/tooltip-handler/canvas-tooltip-handler.ts [hunk 1]
```diff
@@ -55,6 +60,22 @@ export class CanvasTooltipHandler extends BaseTooltipHandler {
...
+    if (!changePositionOnly || isNil(this._attributes)) {
+      const chartTheme = this._chartOption?.getTheme() ?? {};
+      this._attributes = getTooltipAttributes(actualTooltip, this._component.getSpec(), chartTheme);
+    }
+    const { panel, panelDomHeight } = this._attributes ?? {};
+    // canvas模式下, size需要考虑border size, 目的是为了精准判断边界是否超出画布，达到confine效果
+    // html模式不提供confine, 所以不考虑精准计算size
+
+    return {
+      width: panel?.width + panel.lineWidth,
```
Comment (xiaoluoHe) [inline packages/vchart/src/plugin/components/tooltip-handler/canvas-tooltip-handler.ts:74]: panel?.lineWidth
后面也是，`?.`都加一下，或者给 panel 默认值？

File: packages/vchart/src/plugin/components/tooltip-handler/constants.ts [hunk 2]
```diff
@@ -12,15 +14,6 @@ export const DEFAULT_OPTIONS = {
...
    * Y offset.
    */
-  offsetY: 10,
-
-  /**
-   * HTML sanitizer function that removes dangerous HTML to prevent XSS.
-   *
-   * This should be a function from string to string. You may replace it with a formatter such as a markdown formatter.
-   */
-  sanitize: escapeHTML
```
Comment (xiaoluoHe) [inline packages/vchart/src/plugin/components/tooltip-handler/constants.ts:22]: 编码的逻辑去掉了吗？

File: packages/vchart/src/plugin/components/tooltip-handler/dom-tooltip-handler.ts [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    }
+  }
+
+  protected _initStyle() {
+    const tooltipSpec = this._component.getSpec() as ITooltipSpec;
+
+    this._domStyle = getDomStyle(tooltipSpec);
+  }
+
+  // protected _updateDomString(actualTooltip?: ITooltipActual) {
```
Comment (xiaoluoHe) [inline packages/vchart/src/plugin/components/tooltip-handler/dom-tooltip-handler.ts:179]: 这里删掉？

File: packages/vchart/src/plugin/components/tooltip-handler/constants.ts (hunk 2 above)
Comment (xile611) [reply packages/vchart/src/plugin/components/tooltip-handler/constants.ts:22]: 这个地方就改了个名字，我直接在用的地方引用了
这里传来传去了，绕晕了

File: packages/vchart/src/plugin/components/tooltip-handler/canvas-tooltip-handler.ts (hunk 1 above)
Comment (xile611) [reply packages/vchart/src/plugin/components/tooltip-handler/canvas-tooltip-handler.ts:74]: panel不可能为空，我吧 `?.` 都去掉了

**Review by xiaoluoHe — APPROVED** [review]

**Pull request merged**